### ISSUE TITLE
Cache

### ISSUE THREAD

**Issue by RichardLindhout** [PlayForm/Compress#49]
It would be nice if there would a cache folder where the already compressed images are in so they won't need a new compression. It takes long to do the images on my website currently so deployment is slower.

**Comment by nikolaxhristov** [comment]
@RichardLindhout I already though about that! 😁 But that would require some harder thinking.

**Comment by nikolaxhristov** [comment]
But expect it in `1.2.0`

**Comment by RichardLindhout** [comment]
@NikolaRHristov does this work now, would be really cool!

**Comment by NikolaRHristov** [comment]
Hey @RichardLindhout, still not I'm afraid 😅 That's why `astro-compress` is still version 1.1.x 🤣

**Comment by NikolaRHristov** [comment]
I have some time over the weekend so I'll give it a crack.

**Comment by RichardLindhout** [comment]
Wow that would be so great!!

**Comment by lyleschemmerling** [comment]
Hey @NikolaRHristov Thanks again for opening that PR against FusionAuth!

I know you are trying to pull back from this project, but did you have thoughts on how this would work? I might hack on this and make a PR if you haven't gotten very far.

**Comment by NikolaRHristov** [comment]
Yeah, no problem! I've been thinking about using the internal `git` cache of a repository and simply outputting a folder with already processed files. Then just calling that folder to see if something's cached or not. The simplest of things. P.S. I'm not backing out just hedging.

sequenceDiagram
    participant User
    participant AutomationLibrary
    participant Cache
    participant Processor

    User->>AutomationLibrary: Input Files
    loop Check Cache
        AutomationLibrary->>Cache: Check if cached
        Cache-->>AutomationLibrary: Cache status
    end
    alt Cache Found
        AutomationLibrary->>Cache: Retrieve from Cache
        Cache-->>AutomationLibrary: Cached Files
    else Cache Not Found
        AutomationLibrary->>Processor: Process Files
        Processor-->>AutomationLibrary: Processed Files
        AutomationLibrary->>Cache: Store in Cache
        Cache-->>AutomationLibrary: Cache Stored
    end
    AutomationLibrary->>User: Output Files

**Comment by NikolaRHristov** [comment]
This would ideally be implemented in `FilesPipe` - https://github.com/NikolaRHristov/FilesPipe

**Comment by NikolaRHristov** [comment]
This is very well still in ideation stage, so all PRs are welcome.

**Comment by azat-io** [comment]
Thank you, I love your package.
Really looking forward to the caching feature.

**Comment by NikolaRHristov** [comment]
> Thank you, I love your package. Really looking forward to the caching feature.

🙏🏻 Yeah, same, this is a hard feature.

**Comment by jgabriel98** [comment]
Do we have any intent on tackling on this.
This integration is amazing, but +7 minutes on everybuild on a small landing page seems too much of a hit for long term usage

**Comment by jgabriel98** [comment]
Hey guys, @wlami made a **cached** alternative to PlayForm Compress, but it had bugs and the project got abandoned.

So i decided to fork it, fix it, and maitaint it myself.

You guys can check it out here:
https://github.com/jgabriel98/astro-compress

Feel free to open issues and report bugs!
